internal/tiger/cmd/integration_test.go

```diff
@@ -1307,3 +1307,368 @@ func TestAuthenticationErrorsIntegration(t *testing.T) {
+	})
+
+	t.Run("CreateSourceService", func(t *testing.T) {
+		t.Logf("Creating source service: %s", sourceServiceName)
+
+		output, err := executeIntegrationCommand(
+			t.Context(),
+			"service", "create",
+			"--name", sourceServiceName,
+			"--cpu", "shared",
+			"--wait-timeout", "15m",
+			"--no-set-default",
+			"--output", "json",
+		)
+
+		if err != nil {
+			t.Fatalf("Source service creation failed: %v\nOutput: %s", err, output)
+		}
+
+		extractedServiceID := extractServiceIDFromCreateOutput(t, output)
+		if extractedServiceID == "" {
+			t.Fatalf("Could not extract source service ID from create output: %s", output)
+		}
+
+		sourceServiceID = extractedServiceID
+		t.Logf("Created source service with ID: %s", sourceServiceID)
+	})
+
+	t.Run("InsertTestData", func(t *testing.T) {
+		if sourceServiceID == "" {
+			t.Skip("No source service ID available")
+		}
+
+		t.Logf("Creating test table: %s", tableName)
+
+		// Create table
+		output, err := executeIntegrationCommand(
+			t.Context(),
+			"db", "psql", sourceServiceID,
+			"--", "-c", fmt.Sprintf("CREATE TABLE %s (id INT PRIMARY KEY, data TEXT, created_at TIMESTAMP DEFAULT NOW());", tableName),
+		)
+
+		if err != nil {
+			t.Fatalf("Failed to create test table: %v\nOutput: %s", err, output)
+		}
+
+		t.Logf("Inserting test data into table: %s", tableName)
+
+		// Insert test data
+		output, err = executeIntegrationCommand(
+			t.Context(),
+			"db", "psql", sourceServiceID,
+			"--", "-c", fmt.Sprintf("INSERT INTO %s (id, data) VALUES (1, 'test-row-1'), (2, 'test-row-2'), (3, 'test-row-3');", tableName),
+		)
+
+		if err != nil {
+			t.Fatalf("Failed to insert test data: %v\nOutput: %s", err, output)
+		}
+
+		t.Logf("✅ Test data inserted successfully")
+	})
+
+	t.Run("VerifySourceData", func(t *testing.T) {
+		if sourceServiceID == "" {
+			t.Skip("No source service ID available")
+		}
+
+		t.Logf("Verifying test data in source service")
+
+		output, err := executeIntegrationCommand(
+			t.Context(),
+			"db", "psql", sourceServiceID,
+			"--", "-c", fmt.Sprintf("SELECT * FROM %s ORDER BY id;", tableName),
+		)
+
+		if err != nil {
+			t.Fatalf("Failed to query test data: %v\nOutput: %s", err, output)
+		}
+
+		// Verify all three rows are present
+		if !strings.Contains(output, "test-row-1") {
+			t.Errorf("Expected 'test-row-1' in output, got: %s", output)
+		}
+		if !strings.Contains(output, "test-row-2") {
+			t.Errorf("Expected 'test-row-2' in output, got: %s", output)
+		}
+		if !strings.Contains(output, "test-row-3") {
+			t.Errorf("Expected 'test-row-3' in output, got: %s", output)
+		}
+
+		t.Logf("✅ Source data verified: 3 rows present")
+	})
+
+	t.Run("ForkService", func(t *testing.T) {
+		if sourceServiceID == "" {
+			t.Skip("No source service ID available")
+		}
+
+		t.Logf("Forking service: %s with --now strategy", sourceServiceID)
+
+		output, err := executeIntegrationCommand(
+			t.Context(),
+			"service", "fork", sourceServiceID,
+			"--now",
```

Comment:
after this we should also create another fork with --last-snapshot. I would also test that --last-snapshot works fine before a snapshot exists (i.e. gives an error?)